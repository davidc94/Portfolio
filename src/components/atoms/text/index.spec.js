import React from 'react';
import renderer from 'react-test-renderer';
import StyledText from './index';
import Colors from '../../../utils/colors';

describe('StyledText component', () => {
  it('render default StyledText as expected', () => {
    const render = renderer.create(
      <StyledText>I have nothing to offer</StyledText>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('render StyledText with a different font size', () => {
    const render = renderer.create(
      <StyledText size="large">EXTRA BIG</StyledText>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('render StyledText with a different color', () => {
    const render = renderer.create(
      <StyledText color={Colors.lightBlue}>cool blue</StyledText>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('render StyledText with a different margin', () => {
    const render = renderer.create(
      <StyledText margin="large">wow, so much extra space</StyledText>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });
});