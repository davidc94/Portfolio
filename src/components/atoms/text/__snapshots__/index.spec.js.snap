// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledText component render StyledText with a different font size 1`] = `
<p
  className="sc-bdVaJa wkRKl"
  size="large"
>
  EXTRA BIG
</p>
`;

exports[`StyledText component render default StyledText as expected 1`] = `
<p
  className="sc-bdVaJa jhpExi"
>
  I have nothing to offer
</p>
`;
